### PR TITLE
Add path separator to cache key

### PR DESCRIPTION
**Pull request by Aishwarya-Lad**
Please read [CONTRIBUTING.md](CONTRIBUTING.md) for details on our [code of conduct](CODE_OF_CONDUCT.md), and the process for submitting pull requests to us.

Fixes #

## Proposed Changes
Add path separator '/' while looking for directories to restore.

-
-
-

### Description
The logic of prefix matching was causing a conflict where one cache key is a complete prefix of another, to avoid the clashes adding '/' to the cache key.

### Checklist

- [x] Read the [**CONTRIBUTING**](CONTRIBUTING.md) document.
- [ ] Read the [**CODE OF CONDUCT**](CODE_OF_CONDUCT.md) document.
- [x] Add tests to cover changes.
- [x] Ensure your code follows the code style of this project.
- [ ] Ensure CI and all other PR checks are green OR
    - [ ] Code compiles correctly.
    - [ ] Created tests which fail without the change (if possible).
    - [ ] All new and existing tests passed.
- [ ] Add your changes to `Unreleased` section of [CHANGELOG](CHANGELOG.md).
- [ ] Improve and update the [README](README.md) (if necessary).
- [ ] Ensure [documentation](./DOCS.md) is up-to-date. The same file will be updated in [plugin index](https://github.com/drone/drone-plugin-index/blob/master/content/meltwater/drone-cache/index.md) when your PR is accepted, so it will be available for end-users at http://plugins.drone.io.

### CODE REVIEW

File: cache/option.go [hunk 1]
```diff
@@ -55,3 +56,10 @@ func WithFailRestoreIfKeyNotPresent(b bool) Option {
 		o.failRestoreIfKeyNotPresent = b
 	})
 }
+
+// WithDisableCacheKeySeparator disables addition of '/' to cache
```
Comment (hemanthmantri) [inline cache/option.go:60]: `disables addition of '/' to the specified cache key.`

File: main.go [hunk 2]
```diff
@@ -319,6 +319,12 @@ func main() {
 			Value:   false,
 			EnvVars: []string{"PLUGIN_FAIL_RESTORE_IF_KEY_NOT_PRESENT"},
 		},
+		&cli.BoolFlag{
+			Name:    "disable-cache-key-separator",
+			Usage:   "Allow addition of separator (/) to cache-prefix. (defaults to false)",
```
Comment (hemanthmantri) [inline main.go:324]: Disable adding of `/` as the cache key suffix.

File: main.go [hunk 3]
```diff
@@ -319,6 +319,12 @@ func main() {
...
+			Value:   false,
+			EnvVars: []string{"PLUGIN_DISABLE_SEPARATOR", "AWS_PLUGIN_DISABLE_SEPARATOR"},
```
Comment (hemanthmantri) [inline main.go:326]: Why `AWS`?

**Review by hemanthmantri — COMMENTED** [review]
Please add description to the PR of what this change is doing.

**Review by jamie-harness — APPROVED** [review]

**Pull request merged**